tests/micro/zephyr/test_zephyr_armv7m.py

```diff
@@ -0,0 +1,293 @@
+            _create_header_file("output_data", np.zeros(shape=output_shape, dtype="float32"), "include", tf)
+
+        project, _ = _build_project(
+            temp_dir,
+            board,
+            west_cmd,
+            lowered,
+            build_config,
+            extra_files_tar=tar_temp_file.name,
+        )
+
+    return project
+
+
+def _run_model(temp_dir, board, west_cmd, lowered, build_config, sample, output_shape):
+
+    project = _generate_project(temp_dir, board, west_cmd, lowered, build_config, sample, output_shape)
+
+    project.flash()
+
+    with project.transport() as transport:
+        timeout_read = 60
+        # _get_message(transport, "#wakeup", timeout_sec=timeout_read)
+        transport.write(b"start\n", timeout_sec=5)
+        result_line = _get_message(transport, "#result", timeout_sec=timeout_read)
+
+    result_line = result_line.strip("\n")
+    result_line = result_line.split(":")
+    result = int(result_line[1])
+    time = int(result_line[2])
+    logging.info(f"Result: {result}\ttime: {time} ms")
+
+    return result, time
+
+
+@tvm.testing.requires_micro
+def test_armv7m_intrinsic(temp_dir, board, west_cmd, tvm_debug):
+    """Testing a ARM v7m SIMD extension."""
+
+    if board not in [
+        "nrf5340dk",
+        "stm32f746xx_disco",
+        "stm32f746xx_nucleo",
+        "stm32l4r5zi_nucleo",
+    ]:
+        pytest.skip(msg="Platform does not support ARM v7m SIMD extenion.")
+
+    model = conftest.ZEPHYR_BOARDS[board]
+
+    build_config = {"debug": tvm_debug}
+
+    this_dir = pathlib.Path(os.path.dirname(__file__))
+    testdata_dir = this_dir.parent / "testdata" / "armv7m"
+
+    relay_mod, params = _open_tflite_model(testdata_dir / "mnist_model_quant.tflite")
+
+    sample, output_shape = _get_test_data(testdata_dir)
+
+    relay_mod_isa = _apply_desired_layout_isa(relay_mod)
+    # kernel layout "HWIO" is not supported by arm_cpu SIMD extension (see tvm\python\relay\op\strategy\arm_cpu.py)
+    relay_mod_no_isa = _apply_desired_layout_no_isa(relay_mod)
+
+    target = tvm.target.target.micro(
+        model, options=["-keys=arm_cpu,cpu", "-link-params=1", "--executor=aot", "--unpacked-api=1", "--interface-api=c"]
+    )
+
+    temp_dir_isa = temp_dir / "isa"
+    temp_dir_no_isa = temp_dir / "noisa"
+
+    os.makedirs(temp_dir_isa, exist_ok=True)
+    os.makedirs(temp_dir_no_isa, exist_ok=True)
+
+    with tvm.transform.PassContext(opt_level=3, config={"tir.disable_vectorize": True}):
+        lowered_isa = relay.build(relay_mod_isa, target, params=params)
+        lowered_no_isa = relay.build(relay_mod_no_isa, target, params=params)
+        result_isa, time_isa = _run_model(temp_dir_isa, board, west_cmd, lowered_isa, build_config, sample, output_shape)
+        result_no_isa, time_no_isa = _run_model(temp_dir_no_isa, board, west_cmd, lowered_no_isa, build_config, sample, output_shape)
+
+    assert result_no_isa == result_isa
+    assert time_no_isa > time_isa
```

Comment:
while this should essentially always pass, i'm wondering if it may be a bit brittle to check in to TVM's mainline CI. in general with µTVM, we'll gauge performance using an offline process (at least for now). it might be better to track performance regressions using that flow rather than this.

Reply:
You should be able to write this test with AoT and using the Corstone-300 FVP or indeed QEMU, the Cortex-M55 is an implementation of the Armv8-M architecture which is a superset of the Armv7-M architecture and thus contains the SMLAD intrinsic. Thus we would be able to get functional testing going using the Corstone-300 FVP or QEMU.

Reply:
@u99127 did we add this to the regression tests yet? I think I just saw the Makefile change, but not sure we have rebuilt the ci_qemu image to include the FVP yet?

Reply:
@u99127 i'm remembering now i'm not sure if we intended to add Corstone to regression. what do you think of using one of the Zephyr QEMU targets?

Reply:
can we use [ARM Cortex-M3 Emulation (QEMU)](https://docs.zephyrproject.org/latest/boards/arm/qemu_cortex_m3/doc/index.html)?

Reply:
@u99127 do you suggest any other CI test we could use? otherwise, we can test on-device for now.

Reply:
@areusch Will test with QEMU 5 shortly... With MPS2-AN521 qemu emulator.

Reply:
Running on QEMU MPS2-AN521 target results both performance time are equal to zero. Escape this from evaluating in this particular test.